internal/cli/atlas/networking/peering/watch.go

```diff
@@ -57,28 +57,28 @@ func (opts *WatchOpts) watcher() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-
-	switch v := result.(type) {
-	case *atlasv2.AwsNetworkPeeringConnectionSettings:
-		return watcherAWS(v), nil
-	case *atlasv2.AzureNetworkPeeringConnectionSettings:
-		return watcherAzure(v), nil
-	case *atlasv2.GCPNetworkPeeringConnectionSettings:
-		return watcherGCP(v), nil
+	switch v := result.GetProviderName(); v {
+	case "AWS":
+		return watcherAWS(result), nil
+	case "AZURE":
+		return watcherAzure(result), nil
+	case "GCP":
+		return watcherGCP(result), nil
 	}
+	fmt.Print("Network peering changes completed.\n")
 	return false, nil
 }
 
-func watcherGCP(peer *atlasv2.GCPNetworkPeeringConnectionSettings) bool {
-	return *peer.Status == waitingForUser || *peer.Status == failed || *peer.Status == available
+func watcherGCP(peer *atlasv2.BaseNetworkPeeringConnectionSettings) bool {
+	return peer.GetStatus() == waitingForUser || peer.GetStatus() == failed || peer.GetStatus() == available
 }
 
-func watcherAzure(peer *atlasv2.AzureNetworkPeeringConnectionSettings) bool {
-	return *peer.Status == waitingForUser || *peer.Status == failed || *peer.Status == available
+func watcherAzure(peer *atlasv2.BaseNetworkPeeringConnectionSettings) bool {
+	return peer.GetStatus() == waitingForUser || peer.GetStatus() == failed || peer.GetStatus() == available
```

Comment:
I did ensure we use getters in many places and avoid pointers. Getters will prevent panic errors in the future.